src/npctalk_rules.cpp

```diff
@@ -0,0 +1,518 @@
+{
+    if( !guy ) {
+        debugmsg( "Something has gone very wrong, can't find NPC to set rules for.  Aborting." );
+        last_action = "QUIT";
+        return;
+    }
+
+    ImGui::SetWindowSize( ImVec2( window_width, window_height ), ImGuiCond_Once );
+
+    static bool exporting_rules = false;
+    static bool in_popup = false;
+    if( in_popup ) {
+        rules_transfer_popup( exporting_rules, in_popup );
+        return;
+    }
+
+    ImGui::InvisibleButton( "TOP_OF_WINDOW_KB_SCROLL_SELECTABLE", ImVec2() );
+
+    const hotkey_queue &hotkeys = hotkey_queue::alphabets();
+    input_event assigned_hotkey = input_ptr->first_unassigned_hotkey( hotkeys );
+    input_event pressed_key = input_ptr->get_raw_input();
+
+    draw_colored_text( string_format( _( "Rules for your follower, %s" ), guy->disp_name() ) );
+    ImGui::Separator();
+
+    if( ImGui::Button( _( "Import settings from follower" ) ) ) {
+        exporting_rules = false;
+        in_popup = true;
+        return;
+    }
+    if( ImGui::Button( _( "Export settings to follower" ) ) ) {
+        exporting_rules = true;
+        in_popup = true;
+        return;
+    }
+
+    draw_colored_text( _( "Hotkey:" ) );
+    ImGui::NewLine();
+
+
+    print_hotkey( assigned_hotkey );
+    if( ImGui::Button( _( "Default ALL" ) ) || pressed_key == assigned_hotkey ) {
+        ImGui::SetKeyboardFocusHere( -1 );
+        // TODO: use query_yn here as a safeguard against fatfingering. Can't use it right now,
+        // it destructs the imgui instance(?!) which obviously causes us to crash
+        npc_follower_rules default_values; //Call the constructor and let *it* tell us what the default is
+        guy->rules = default_values;
+    }
+
+    int rule_number = 0;
+    /* Handle all of our regular, boolean rules */
+    for( const std::pair<const std::string, ally_rule_data> &rule_data : ally_rule_strs ) {
+        ImGui::PushID( rule_number );
+        assigned_hotkey = input_ptr->next_unassigned_hotkey( hotkeys, assigned_hotkey );
+        rule_number++;
+        ImGui::NewLine();
+        print_hotkey( assigned_hotkey );
+        const ally_rule_data &this_rule = rule_data.second;
+        bool rule_enabled = false;
+        std::string rules_text;
+        if( guy->rules.has_flag( this_rule.rule ) ) {
+            rules_text = string_format( "%s", get_parsed( this_rule.rule_true_text ) );
+            rule_enabled = true;
+        } else {
+            rules_text = string_format( "%s", get_parsed( this_rule.rule_false_text ) );
+        }
+        if( rule_enabled ) {
+            ImGui::PushStyleColor( ImGuiCol_Button, c_green );
+        }
+        if( ImGui::Button( _( "Toggle" ) ) || pressed_key == assigned_hotkey ) {
+            ImGui::SetKeyboardFocusHere( -1 );
+            guy->rules.toggle_flag( this_rule.rule );
+            guy->rules.toggle_specific_override_state( this_rule.rule, !rule_enabled );
+        }
+        ImGui::PopStyleColor();
```

Comment:
This pop needs to be guarded by the same condition as the push, so that they pair up correctly.